src/main/java/org/jboss/ws/tools/security/legacy/SecurityManagerUtils.java

```diff
@@ -0,0 +1,107 @@
+package org.jboss.ws.tools.security.legacy;
+
+import java.lang.reflect.Method;
+
+/**
+ * A helper class that caches and provides status about the SecurityManager API, which is deprecated starting from
+ * JDK17 (JEP 411), and removed on JDK 23+ (JEP 486).
+ * It also keeps the java.security.* deprecated APIs separate.
+ */
+public class SecurityManagerUtils {
+    // Cache reflection based SecurityManager API state to avoid repeated computation
+    private static final boolean SECURITY_MANAGER_AVAILABLE;
+    private static final Method GET_SECURITY_MANAGER_METHOD;
+
+    static {
+        boolean available = false;
+        Method method = null;
+        try {
+            // Does getSecurityManager exist?
+            method = System.class.getMethod("getSecurityManager");
```

Comment:
This method have been marked for removal but it is still present in JDK 25 LTS.
According to https://docs.oracle.com/en/java/javase/24/security/security-manager-is-permanently-disabled.html
it was disabled in JDK24 including so I would propose check something like:

available = Runtime.version().feature() < 24;